cmd/root.go

```diff
@@ -47,104 +46,168 @@ var RootCmd = &cobra.Command{
 
 		utils.SetDockerEngine(eng)
 
-		img1Arg := args[0]
-		img2Arg := args[1]
-		diffArgs := []string{}
-		allDiffers := getAllDiffers()
-		for _, name := range allDiffers {
-			if *diffFlagMap[name] == true {
-				diffArgs = append(diffArgs, name)
+		analyzeArgs := []string{}
+		allAnalyzers := getAllAnalyzers()
+		for _, name := range allAnalyzers {
+			if *analyzeFlagMap[name] == true {
+				analyzeArgs = append(analyzeArgs, name)
 			}
 		}
-		// If no differs are specified, perform all diffs as the default
-		if len(diffArgs) == 0 {
-			diffArgs = allDiffers
+
+		// If no differs/analyzers are specified, perform them all as the default
+		if len(analyzeArgs) == 0 {
+			analyzeArgs = allAnalyzers
 		}
 
-		var wg sync.WaitGroup
-		wg.Add(2)
+		if len(args) == 1 {
```

Comment:
I actually think this structure is fine. We should probably at least put a comment here saying semantically what each case means (`analyze` = one image, `diff` = two), and then put something about it in the usage/`--help` command. I do like the tool choosing which path to take based on how many images you provide it though, seems intuitive to me